clang/test/SemaSYCL/unnamed-kernel.cpp

```diff
@@ -117,8 +110,7 @@ struct MyWrapper {
 int main() {
   cl::sycl::queue q;
 #ifndef __SYCL_UNNAMED_LAMBDA__
-  // expected-error-re@Inputs/sycl.hpp:233 {{'(lambda at {{.*}}unnamed-kernel.cpp{{.*}}' is an invalid kernel name type}}
-  // expected-note@Inputs/sycl.hpp:233 {{unnamed type used in a SYCL kernel name}}
+  // expected-error@Inputs/sycl.hpp:233 {{type cannot be unnamed}}
```

Comment:
Is it clear from the context what 'type' is being talked about here?  The previous message at least used to refer to a lambda.

Reply:
Previously we used to print the "type" of the problematic kernel name, but we now print "declaration" of the type. 
Since, kernel names are empty for this case, there is no declaration to print.
I agree that it is not very obvious what 'type' we are referring to here. Maybe changing the text to - kernel name is unnamed/empty/missing would help?
Please let me know if there are better suggestions.

Reply:
I think something like 'type cannot be unnamed; unnamed lambda used' may be sufficient?  Could you detect when the diagnostic is issued that this is due to an unnamed lambda?

Reply:
I prefer to drop the word 'type' since we are no longer dealing with type of the kernel name.
I have updated the diagnostic to :` "unnamed lambda <print-unnamed-lambda> used"`
Example:
`unnamed lambda '(lambda at /iusers/sundaram/forked-repo/llvm/clang/test/SemaSYCL/unnamed-kernel.cpp:119:54)' used`